control-plane/gateways/deployment_dataplane_container.go

```diff
@@ -107,6 +114,13 @@ func consulDataplaneContainer(config GatewayConfig, resources *corev1.ResourceRe
 		ContainerPort: int32(constants.DefaultWANPort),
 	})
 
+	if containerConfig != nil {
```

Comment:
I think this should be in an `if-else` with the port above.

if containerConfig, define the port named `wan` using the `PortModifier`
if nil containerConfig, define the port named `wan` using the `DefaultWANPort`